mono/metadata/metadata.h

```diff
@@ -309,6 +309,11 @@ typedef struct {
 	unsigned int token    : 31;
 } MonoCustomMod;
 
+typedef struct {
+	MonoCustomMod exported;
+	void *image;
+} MonoCustomModInternal;
```

Comment:
If you define it here, it's not internal. It's part of the public API.

Reply:
We shouldn't store then on each cmod, but change MonoType to have an optional cmod container.

IE, change MonoType to this:
```c 
struct _MonoType {
	union {
		MonoClass *klass; /* for VALUETYPE and CLASS */
		MonoType *type;   /* for PTR */
		MonoArrayType *array; /* for ARRAY */
		MonoMethodSignature *method;
		MonoGenericParam *generic_param; /* for VAR and MVAR */
		MonoGenericClass *generic_class; /* for GENERICINST */
	} data;
	unsigned int attrs    : 16; /* param attributes or field flags */
	MonoTypeEnum type     : 8;
	unsigned int has_cmods : 1;  /* has a cmod appended at the end */
	unsigned int byref    : 1;
	unsigned int pinned   : 1;  /* valid when included in a local var signature */
};
```

Then have this:
```c
typedef struct CustomModContainer {
   int count;
   MonoImage *image;
   MonoCustomMod modifiers [1]; /*the actual size of this array is count*/
}
```

You'd be able to fetch it using something like this:
```c

CustomModContainer*
mono_type_get_cmods (MonoType *t) {
  if (!t->has_cmods)
    return null;
  return (CustomModContainer*)(((char*)t) + sizeof (MonoType));

}
```

Reply:
Maybe a bit nicer:

```c
typedef struct _MonoTypeWithModifiers {
  MonoType type;
  CustomModContainer cmods;
} MonoTypeWithWrappers;
```

So you can just do `&((MonoTypeWithModifiers*)type)->cmods` in `mono_type_get_cmods`

Reply:
That looks good @lambdageek, that works just like our other prefixed struct types